vehicle/remote_control_process.py

```diff
@@ -246,1086 +306,939 @@ def close_motor_socket(self):
+            self.autonomy_hold = True
+            self.activate_autonomy = False
+            self.control_state = CONTROL_OVERRIDE
+
+        (user_web_page_plot_steer_cmd, user_web_page_plot_strafe_cmd,
+         strafe_d, steer_d, strafe_p, steer_p,
+         autonomy_vel_cmd, autonomy_steer_cmd, autonomy_strafe_cmd
+         ) = self.calc_commands_for_autonomy(calculated_rotation, calculated_strafe, drive_reverse)
+
+        # In the following list the order for when we set control state
+        # matters
+        # TODO: clarify? What does this note mean?
+
+        time7 = time.time() - debug_time
+
+        error_messages, fatal_error, zero_output = self.safty_checks(absolute_path_distance, gps_path_angle_error)
+
+        if self.loop_count % _ERROR_SKIP_RATE == 0:
+            for error in error_messages:
+                self.logger.error(error)
+
+        self.maybe_restart_wifi()
+
+        if zero_output:
+            if self.activate_autonomy and time.time() - self.disengagement_time > _DISENGAGEMENT_RETRY_DELAY_SEC:
+                self.disengagement_time = time.time()
+                self.logger.error("Disengaging Autonomy.")
+                self.write_errors(error_messages)
+                if fatal_error:
+                    self.autonomy_hold = True
+                    self.activate_autonomy = False
+        elif not self.activate_autonomy:
+            self.resume_motion_timer = time.time()
+            self.control_state = CONTROL_ONLINE
+
+        time8 = time.time() - debug_time
+
+        # Activate a brief pause at the end of a track.
+        if time.time() - self.load_path_time < _PATH_END_PAUSE_SEC and not zero_output:
+            zero_output = True
+            # Don't use the motion timer here as it reactivates alarm.
+            if self.loop_count % 10 == 0:
+                self.logger.info("Taking a short delay at the end of the path so zeroing out autonomy commands.")
+
+        if time.time() - self.disengagement_time < _DISENGAGEMENT_RETRY_DELAY_SEC:
+            zero_output = True
+            self.resume_motion_timer = time.time()
+            if self.loop_count % 10 == 0:
+                self.logger.info(
+                    "Disengaged for {:.1f} more seconds so zeroing out autonomy commands."
+                    .format(_DISENGAGEMENT_RETRY_DELAY_SEC - (time.time() - self.disengagement_time)))
+
+        if (self.activate_autonomy and not zero_output and
+                time.time() - self.resume_motion_timer < _RESUME_MOTION_WARNING_TIME_SEC):
+            zero_output = True
+            self.alarm1.value = True
+            self.alarm2.value = False
+            self.alarm3.value = True
+        else:
+            self.alarm1.value = False
+            self.alarm2.value = False
+            self.alarm3.value = False
+
+        # Determine final drive commands.
+        vel_cmd, steer_cmd, strafe_cmd = self.calc_drive_commands(
```

Comment:
extract method. the method name is just arbitrarily chosen. lots of room to improvement.